ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKSourceFactory.java

```diff
@@ -48,12 +48,14 @@ private SchemaKSourceFactory() {
   public static SchemaKStream<?> buildSource(
       final KsqlQueryBuilder builder,
       final DataSource<?> dataSource,
-      final LogicalSchemaWithMetaAndKeyFields schemaWithMetaAndKeyFields,
       final QueryContext.Stacker contextStacker,
       final Optional<AutoOffsetReset> offsetReset,
       final KeyField keyField,
       final SourceName alias
   ) {
+    final LogicalSchemaWithMetaAndKeyFields schemaWithMetaAndKeyFields =
+        LogicalSchemaWithMetaAndKeyFields.fromOriginal(alias, dataSource.getSchema());
```

Comment:
Adding ROWTIME / ROWKEY has moved down to here. Which is what?  The physical plan, right?

Reply:
Yeah. Going to move this down to the plan-building layer very soon and get rid of `LogicalSchemaWithMetaAndKeyFields`.

Reply:
Sa----weet!